pycode/memilio-surrogatemodel/memilio/surrogatemodel_test/test_surrogatemodel_ode_secir_simple.py

```diff
@@ -100,110 +105,402 @@ def test_data_generation_save(self):
+        self.assertEqual(str(error.exception), error_message)
 
-        model_mlp_multi_input_single_output = network_architectures.mlp_multi_input_single_output()
-        self.assertEqual(len(model_mlp_multi_input_single_output.layers), 5)
+        with self.assertRaises(ValueError) as error:
+            network_architectures.mlp_multi_input_single_output(
+                1, 1, 0
+            )
+        error_message = "Number of neurons per layer must be at least 1, here 0"
+        self.assertEqual(str(error.exception), error_message)
+
+        model = network_architectures.mlp_multi_input_single_output(3, 4, 84)
+        self.assertEqual(len(model.layers), 7)
         input_zero = np.zeros((1, 5, 8))
-        output_zeros = model_mlp_multi_input_single_output(input_zero)
+        output_zeros = model(input_zero)
         self.assertEqual(output_zeros.shape[0], 1)
         self.assertEqual(output_zeros.shape[1], 1)
-        self.assertEqual(output_zeros.shape[2], 8)
+        self.assertEqual(output_zeros.shape[2], 3)
+
+# Testing mlp_multi_input_multi_output
+    def test_mlp_multi_single(self):
+        with self.assertRaises(ValueError) as error:
+            network_architectures.mlp_multi_input_multi_output(
+                0, 12, 12, 12
+            )
+        error_message = "label width has to be a positive integer"
+        self.assertEqual(str(error.exception), error_message)
+
+        with self.assertRaises(ValueError) as error:
+            network_architectures.mlp_multi_input_multi_output(
+                12, 0, 12, 12
+            )
+        error_message = "Output dimension must be at least 1, here 0"
+        self.assertEqual(str(error.exception), error_message)
+
+        with self.assertRaises(ValueError) as error:
+            network_architectures.mlp_multi_input_multi_output(
+                12, 12, -1, 12
+            )
+        error_message = "Number of layers must be at least 0, here -1"
+        self.assertEqual(str(error.exception), error_message)
+
+        with self.assertRaises(ValueError) as error:
+            network_architectures.mlp_multi_input_multi_output(
+                12, 12, 12, 0
+            )
+        error_message = "Number of neurons per layer must be at least 1, here 0"
+        self.assertEqual(str(error.exception), error_message)
 
-        model_lstm_single = network_architectures.lstm_network_multi_input_single_output()
-        self.assertEqual(len(model_lstm_single.layers), 2)
-        input_zero = np.zeros((1, 1, 8))
-        output_zeros = model_lstm_single(input_zero)
+        model = network_architectures.mlp_multi_input_multi_output(
+            12, 3, 4, 84)
+        self.assertEqual(len(model.layers), 7)
+        input_zero = np.zeros((1, 5, 8))
+        output_zeros = model(input_zero)
+        self.assertEqual(output_zeros.shape[0], 1)
+        self.assertEqual(output_zeros.shape[1], 12)
+        self.assertEqual(output_zeros.shape[2], 3)
```

Comment:
in general the tests are great :) I think 1-2 tests where we test performance and not just structure would be nice.